src/Common/tests/System.Net/TaskAPMExtensions.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+/// <summary>
+/// Task APM helper based on http://blogs.msdn.com/b/pfxteam/archive/2011/06/27/10179452.aspx
+/// </summary>
+
+
+namespace System.Threading.Tasks
+{
+    public static class TaskAPMExtensions
+    {
+        public static Task ToApm(this Task task, AsyncCallback callback, object state)
+        {
+            if (task.AsyncState == state)
+            {
+                if (callback != null)
+                {
+                    task.ContinueWith(delegate { callback(task); },
```

Comment:
Instead of changing this, I've added a TODO to completely remove this and rewrite the tests using TPL: #3144

This is a temporary workaround to avoid rewriting and stabilizing all tests using TPL instead of APM.